### ISSUE TITLE
PorygonA Updates

### ISSUE THREAD

**Issue by Sharlikran** [ESOUIMods/LoreBooks#7]
@PorygonA Okay lets just discuss in an issue rather then the comments section now that you are set up.

So the `px` and py` are the LibGps coordinates for the x, y location of the book. The tag `pm` is the primary map ID for the pin. This means it is where you are standing when you get the information and the location the book is to be displayed primarily.

`[7114] = { ["e"] = { [1] = { ["px"] = 0.0815220019, ["py"] = 0.5830836017, ["pm"] = 2114, }, }, }, { ["pnx"] = 0.7893669605, ["pny"] = 0.4129191339 }, -- Green Serpent Bounty, systres/u34_systreszone_base_0`

The reason for the `[7114] = { ["e"] = {` and the rest is if it's a new book I have not entered yet. So what you really want is basically this part.

`[1] = { ["px"] = 0.0815220019, ["py"] = 0.5830836017, ["pm"] = 2114, },`

So essentially that is all you are adding. Then comes the hard part. We will use the book you want to update.

```
  [7114] = {
    ["cn"] = "Systres Tomes and Scrolls",
    ["n"] = "Green Serpent Bounty",
    ["e"] = {
    },
    ["c"] = false,
  },
```
The above is what you start with. Then you paste in the line I mentioned.

## Step one
```
  [7114] = {
    ["cn"] = "Systres Tomes and Scrolls",
    ["n"] = "Green Serpent Bounty",
    ["e"] = {
      [1] = { ["px"] = 0.0815220019, ["py"] = 0.5830836017, ["pm"] = 2114, },
    },
    ["c"] = false,
  },
```
Add the line for the book.

## Step two
```
  [7114] = {
    ["cn"] = "Systres Tomes and Scrolls",
    ["n"] = "Green Serpent Bounty",
    ["e"] = {
      [1] = { ["px"] = 0.0815220019, ["py"] = 0.5830836017, ["pm"] = 2114, },
    },
    ["c"] = true,
  },
```
Change `["c"]` to `true` this means it has data in it.

Then you have to be a human. Is this on an overland map? If it is only an overland book you are done. If not now you have to ask are you in a city?

For the city remember you are in it at the time you get the book. So your map ID might be 2114 for example for that city. The fake pin slash command could return 10 as an example. I don't know what High Isle is but we will use 9999 to avoid confusion for what it really is. So with that pretend ID number this is how a pin looks when you are entering a pin for a city.

## Step three, add the parent zone map ID
```
  [7114] = {
    ["cn"] = "Systres Tomes and Scrolls",
    ["n"] = "Green Serpent Bounty",
    ["e"] = {
      [1] = { ["px"] = 0.0815220019, ["py"] = 0.5830836017, ["pm"] = 2114, ["zm"] = 9999, },
    },
    ["c"] = true,
  },
```
Now with that it is complete. You have the pin location while you are standing in the city px, py, and pm and you have told Lorbooks that the parent zone is 9999 by adding that with `["zm"] = 9999,`. When you are looking at the entire zone Lorebooks will know how to draw the book in both places.

## Dungeon example
```
  [2011] = {
    ["c"] = true,
    ["cn"] = "Lore and Culture",
    ["n"] = "Mastery of Discipline",
    ["q"] = 4959,
    ["e"] = {
      [1] = { ["px"] = 0.2941044142, ["py"] = 0.3928760078, ["pm"] = 420, ["zx"] = 0.2927644048, ["zy"] = 0.3921875965, ["zm"] = 20, ["d"] = true, },
    },
  },
```
So this example is a dungeon pin. px, py and pm are the same. It is where you are when you are underground in the dungeon. So what is new is the `["d"]` because that means you are in the dungeon since it is set to true. That prevents the pin from being drawn on the overland map off in some weird location you can not walk to.

What makes it draw on the overland map correctly is the zx, zy, and zm. They are the x and y locations of the pin where you are when you get the fake pin information. Which you normally get at the entrance to the delve. Dungeon usually means anything not a city or the overland map like a delve or undercroft.

So that is why the dungeon pin has two sets of coordinates. One set for where you are in the dungeon and one set for the overland map so people see a pin at about the entrance to the Delve.

**Comment by Sharlikran** [comment]
Now commit: https://github.com/ESOUIMods/LoreBooks/commit/46eb008815d622396448f40daef9abbdfd731a6a is the first change as long as it is just an overland book. Now add the `["zm"]` if it is in a city, or the other data if it is in a Delve. Then update the commit on your end to your branch rather then `master`

**Comment by PorygonA** [comment]
So the one I'm working on

```
  [7033] = {
    ["cn"] = "Systres Tomes and Scrolls",
    ["n"] = "Report for the Head Jailer",
    ["e"] = {
      [1] = {
        ["px"] = 0.0763844033, ["py"] = 0.5852776124, ["pm"] = 2114, ["zm"] = , },
    },
    ["c"] = true,
  },
```

2114 is the High Isle zone number. How exactly do I find the zm for Amenos Station, the name of the city this is found in?

**Comment by Sharlikran** [comment]
Okay so this is one that may or may not need an extra note. I'm going to say it doesn't but I'm still going to show you anyway. The notes however are not complete sentences so they can fit on the tooltip. For example you don't say, "Go around the corner up the stairs to the Green Plant by the sleeping cat." That would be too long.

First when you get the coordinates there are three slash commands

- /lbpos : LBPOS for Lorebooks Position
- /lbfake : for a fake Eidetic Memory location icon when the Lorebook is not on the main zone and you have to enter a delve or public dungeon
- /lbfakebook : for a fake Shalidor's Library location icon when the Lorebook is not on the main zone and you have to enter a delve or public dungeon

For this one it is `/lbpos` and `/lbfake`

![image](https://user-images.githubusercontent.com/1912426/208941113-e6bdf331-f2e3-4ff3-b2b9-1090330dc997.png)

That is the Lorebook to be recorded

![image](https://user-images.githubusercontent.com/1912426/208941263-d965dae3-ae7e-46b3-8037-08eefe4818fe.png)

I will refer to this later but notice how I am not really in the building. Your player Arrow is outside the building, sorta. You are underground when you are in the Jailers' Keep. This is a weird concept but something you need to be familiar with as you need to comprehend that as you make fake pins for places where it is more complicated.

This is where I am currently. The primary location for the book. It is always where you are standing. It could be that you are in a Zone, Dungeon, Delve, Undercroft, etc. but try to remember it is where you are physically standing.

When you use `/lbpos` there is a special command that forces the game to update its internal map information to where you are standing to make sure the location information is accurate. If I didn't do that then because the developers really don't need that there could be information from another location you were at recently.

Knowing that technical stuff isn't needed but I hope it helps you understand why `/lbpos` behaves the way it does. It can change the map you have selected and if you don't know why then it could be confusing.

Then we type the command to get the information with the book open. If not I could not get the information for the book.

![image](https://user-images.githubusercontent.com/1912426/208943125-eab1d6a4-91d3-415c-aacf-65679c12a671.png)

![image](https://user-images.githubusercontent.com/1912426/208944087-18e2ecf2-2a59-4e55-9bff-b40170123e02.png)

Then you see the text in chat which would end up being this.

`[7033] = { ["e"] = { [1] = { ["px"] = 0.0763696033, ["py"] = 0.5852916124, ["pm"] = 2214, }, }, }, { ["pnx"] = 0.5842999220, ["pny"] = 0.3729238212 }, -- Report for the Head Jailer, systres/u34_amenosstation_city_base_0`

There is several things there but what we want for now is the `[1]` part and we know the book ID is 7033.

```
  [7033] = {
    ["cn"] = "Systres Tomes and Scrolls",
    ["n"] = "Report for the Head Jailer",
    ["e"] = {
      [1] = { ["px"] = 0.0763696033, ["py"] = 0.5852916124, ["pm"] = 2214, },
    },
    ["c"] = true,
  },
```
So where you are now is map ID 2214.

![image](https://user-images.githubusercontent.com/1912426/208944776-d83cd64d-788d-47b5-80b9-523098366c76.png)

You don't need that extra window to do this. It's all done for you with the slash command.

Now you need the fake pin information. If it were a Dungeon you enter it differently but for a city where you are standing works for both the city map and the overland map. That's why for a city you only need the other map ID from the parent map. Call it what you want but it's a Map ID. There are other things like a ZoneId or mapIndex but those are different.

To get that you have to zoom the map out first.

![image](https://user-images.githubusercontent.com/1912426/208945385-6a66c337-2c0f-4502-8188-10ed2efb2cdf.png)

When you use `/lbfake` it won't zoom the map into where you are standing so it will get the information for where you are standing in relation to the map you the player have selected. This would be required for a City like Vvardenfel where you have the Map ID for the Zone itself, the map ID for the city, and the map ID for where you are inside the buildings. For example the Fighters Guild. Those need three or sometimes 4 map pins. 

That is why `/lbfake` gives you the information where you are for the map you selected because once entered it will put the book's Map Pin where you feel it should go to direct the user to it.

`[fake] = { ["e"] = { [1] = { ["px"] = 0.0763696035, ["py"] = 0.5852916015, ["pm"] = 2114, ["fp"] = true, }, }, }, { ["pnx"] = 0.6805444360, ["pny"] = 0.4595537484 }, -- fake Eidetic Memory location, systres/u34_systreszone_base_0`

Now notice the few differences. First it says `[fake]` and the map texture is at the end. Most of that is there so I know where the player is from what they copy paste into the comments. The part you want, which will initially be confusing is the `["pm"]` because that is where you are standing for the map you selected. So the mapID for the overland map is 2114. But I have to have a unique letter for the data files so it's `["zm"]` for the zone's map.

```
  [7033] = {
    ["cn"] = "Systres Tomes and Scrolls",
    ["n"] = "Report for the Head Jailer",
    ["e"] = {
      [1] = { ["px"] = 0.0763696033, ["py"] = 0.5852916124, ["pm"] = 2214, ["zm"] = 2114, },
    },
    ["c"] = true,
  },
```

The numbers are similar so that can add to the confusion however, that is the correct way to get the pin entered for the City.

## Extra Locations Details 

Just know this is an option but something I don't feel you need to do on a regular basis and we don't for this book. It's an option for things that are super complicated to find. 

This information is under the Language folder and then the respective language like EN. You have to enter the English value for all Languages regardless of whether or not it is translated or there will be an error.

```
--[[These are added as Location Details or ["ld"]
]]--
LoreBooks.locationDetails = {
  [1] = "Guest Room",
  [2] = "Attic Room",
  [3] = "Hidden Basement",
}
```
So what would have to happen is another new note would be added and you might make a comment. I haven't because I would normally be the only person.

```
LoreBooks.locationDetails = {
  [1] = "Guest Room",
  [2] = "Attic Room",
  [3] = "Hidden Basement",
  [4] = "Jailers' Keep",
}
```

Then the complete pin information for the extra details
```
  [7033] = {
    ["cn"] = "Systres Tomes and Scrolls",
    ["n"] = "Report for the Head Jailer",
    ["e"] = {
      [1] = { ["px"] = 0.0763696033, ["py"] = 0.5852916124, ["pm"] = 2214, ["zm"] = 2114, ["ld"] = 4, },
    },
    ["c"] = true,
  },
```
![image](https://user-images.githubusercontent.com/1912426/208949088-adaea983-983d-473c-a813-fc5b8b38e362.png)

Then you have the additional note so you aren't trying to look in the bushes or on a large rock for the book. The reason being there is no small map for the Jailers' Keep yet you will sorta zone under the world in the city into the Jailers' Keep so that's why your player Arrow is off outside the building where you could never really walk to.

I still wouldn't add the extra details but that would be how you get the City pin, how you enter it, how to get the fake pin and use that for the `["zm"]` portion, and if it is really needed an extra note to help the user find the book. I prefer we talk about the notes first though.

**Comment by Sharlikran** [comment]
```
    ["e"] = {
      [1] = { ["px"] = 0.0763696033, ["py"] = 0.5852916124, ["pm"] = 2214, ["zm"] = 2114, },
    },
```
I know it's nit-picky but enter all the information on one line. I ask that because when you have say 4 pin locations for a complicated book then I want it that way so that a year form now it's easier to remember which lines belong together for a book. Because you can enter comments at the end like `-- Book Name` and then everyone would know the 4 lines are for the same book.

**Comment by Sharlikran** [comment]
@PorygonA After the holidays keep in touch and I can answer any questions. In the forums I can be a bit abrupt but for learning something when I know it is new to you I'm happy to help. Especially if you have trouble with Git For Windows and making updates.

**Comment by PorygonA** [comment]
Will do.

I've been starting with the books that I've entered the data for, and auditing to make sure they show up on the map. What is the ["zt"]?

An example:

`  [5450] = {
    ["c"] = true,
    ["e"] = {
      [1] = {
        ["d"] = true,
        ["py"] = 0.6902691936,
        ["zt"] = 1121,
        ["px"] = 0.5083984238,
        ["pm"] = 1655,
      },
    },
    ["n"] = "Letter to Bakul",
    ["cn"] = "Anequina Archives",
  },`

**Comment by Sharlikran** [comment]
That is more or less unused. It is for special cases where I could not convert the old data with the information provided.

If you have a book, if the old data has that but you have all the new data, only enter the new data please.

**Comment by Sharlikran** [comment]
Not accusing anyone but I do want to mention this to each of you because there are a few books this way. I just don't remember if there was support for the Reach in v23 or not. Plus I have been the only person entering data so far. So I don't see how this got this way.

The data from what is copy pasted into chat is all you need except you may need to rename `["pm"]` to `["zm"]` for fake pins.

This is incorrect
```
  [6447] = {
    ["c"] = true,
    ["e"] = {
      [1] = {
        ["py"] = 0.2792400000,
        ["px"] = 0.3814500000,
        ["pm"] = 1814,
        ["mn"] = 42,
        ["d"] = false,
        ["r"] = false,
        ["i"] = 12,
      },
    },
    ["n"] = "Namira's Dance",
    ["cn"] = "The Reach Reader",
  },
```
This is all that is needed
```
  [6447] = {
    ["c"] = true,
    ["e"] = {
      [1] = { ["px"] = 0.3814180048, ["py"] = 0.2793443956, ["pm"] = 1814, },
    },
    ["n"] = "Namira's Dance",
    ["cn"] = "The Reach Reader",
  },
```

**Comment by Sharlikran** [comment]
@PorygonA and use your branch [dev-porygona](https://github.com/ESOUIMods/LoreBooks/tree/dev-porygona). Switch to it in Git For Windows. I may rebase and if I do just clone the repo again. You might wan to work on your book entries in the `.\live\Addons` folder so you see the changes. Then use any free program or a program like [BeyondCompare](https://www.scootersoftware.com/) known as a `diff program`. That way you copy your changes into whatever is current.

I'm not sure how to keep your branch up to date with you being new to Git so not sure how to handle things the best.

Another thing we could do is let me know when you have something to submit and I can rebase the branch first and then you can copy your changes using a diff program so you don't revert other changes. Then I can look at the changes on your branch.

It might take some trial and error to get a good workflow.

Again enjoy the Holidays this can wait.

**Comment by PorygonA** [comment]
I've only touched High Isle, Rockgrove, and Sunspire. No clue about the Reach thing.

I've been working on my book entries in the \Live\Addons folder, no worries there. I gotta make sure everything actually shows up properly.

I'll have the High Isle + Ascending Tide up soon.

**Comment by Sharlikran** [comment]
Okay I'll get your branch ready. I appreciate you understanding about what I said. There was no tonality in it or accusation. I'm typing in a browser and no way to position myself to be sure you know it's just for clarification. Thanks.

**Comment by Sharlikran** [comment]
Because I'm updating the routines to draw the pins and I found this.
```
  [6395] = {
    ["c"] = true,
    ["e"] = {
      [1] = {
        ["px"] = 0.3426200000,
        ["py"] = 0.2874700000,
        ["pm"] = 1858,
        ["mn"] = 42,
        ["d"] = false,
        ["r"] = false,
        ["i"] = 12,
        ["zm"] = 1814,
      },
    },
    ["n"] = "An Imperial in Markarth",
    ["cn"] = "The Reach Reader",
  },
```
And I'm like, how? If I did that in the beginning and that was a long time ago when I didn't understand how things worked.

**Comment by Sharlikran** [comment]
```
  [5899] = {
    ["c"] = true,
    ["q"] = 6480,
    ["e"] = {
      [1] = { ["px"] = 0.7014312166, ["py"] = 1.5382663980, ["pm"] = 1809, ["zx"] = 0.7014312222, ["zy"] = 1.5382664378, ["zm"] = 1747, ["d"] = true, ["qp"] = true, },
    },
    ["n"] = "Miner's Journal",
    ["cn"] = "Solitude Soliloquies",
  },
```
`["q"] = QuestID` to be used for the name and whether or not you are actively doing the quest or whether or not you completed the quest.
`["qp"] = true,` Quest In Progress. Means you have to have the quest is your Journal for the book to be shown.
`["qc"] = true,`  = Quest Completed. Means you have to have completed the quest for the book to be shown.

`["i"]` and `["l"]` have been depreciated.

**Comment by Sharlikran** [comment]
@PorygonA The current syntax should be updated for the [Readme](https://github.com/ESOUIMods/LoreBooks#eidetic-memory-data-syntax).

**Comment by PorygonA** [comment]
Gotcha, looks great!

**Question**: What exactly should I be doing right now? Yachoor seems to have uploaded all of the data that I posted in the thread, so what exactly needs to be done still?

**Comment by Sharlikran** [comment]
I had no idea anyone was helping and I would bet he did not think anyone would try to help. I don't think anyone would have. I thought I would have to do everything after the new year like I was planning.

However, I doubt everything has been found or is updated enough.

This is why I'm so picky about how things are entered. Because you could just go to a zone on an alt char that has not done any books and look for books for that character. As you go if you see the new information with one line like I have been suggesting then it's fine, just look at something else. If not and the old format is there then update that.

It's volunteer work anyway. Approach the situation that way. If you feel everything is updated and fine then that's fine you don't have to do anything. I'm pretty laid back about all this. Next year at some point there will be a new zone. 

I would also think there are still books that have not been discovered since The Reach because that was the record above I was confused about. I bet that was just part of the conversion process and I had no idea how things worked. Although It's all changed now. I just don't think everything is completed but I really don't have a way to generate a report. So looking at the one line entries vs the multiple line entries is really it. Yachoor did make some that were two lines or just added the dual pin information as one line. Having the `["zm"]` is an indicator it's updated at least a little bit.

See where things take you. If you are just out playing and you see something record it and go over it at a later time. Just let me know so I can update your branch before you copy anything to it so it doesn't override things already updated. I can pick though it and go form there.

**Comment by Sharlikran** [comment]
Another thing as well. There are going to be books you can only see while on the quest. There are also books devs place outside the instance because after you do the quest you can not go there anymore. There has never been a way to handle that. ZOS added two new functions recently that allow me to check if the player is on a quest by the quest ID and if they have completed it without the dependency of LibQuestData. 

I know there are some books in Greymoor keep that I have pin locations for but I do not have accurate definitions for them. Meaning if you can't read them or they don't even show up on the table, rock, floor or whatever unless you are on the quest then that needs updated. Because it's confusing if you see a pin and you are running around trying to find it only to realize you can't unless you are doing the story quests.

Those kind of fine details are still needed especially since I just refined the syntax to support Quests and things like that. Then you have the Locations Details where if a pin is pretty hard to find we can add something to help with that.

With quests because I have LibQuestData I can look up the Quest ID by the name. So if you are just playing, leveling a char to 50 and you notice a book you could not find and now it's on the map because you are on the quest, well nothing tells you the quest ID. So just stop a moment, write down the quest name and then get the books location and it's fake pin

Awesome Side Quest
Pin
FakePin

If you have all three of those things I can update that. If you want to then just make sure you save the information. If I need to refine it I can and then you can see the difference if I make any changes so you can do the same in the future.

If you don't have Quest Map already get that also and then you have both Lorebooks, Quest Map, and Skyshards so you can just level a character and talk to guildies and have fun.

**Comment by Sharlikran** [comment]
```
  [338] = {
    ["c"] = true,
    ["q"] = 4060,
    ["e"] = {
      [1] = { ["px"] = 0.6301764233, ["py"] = 0.2815471957, ["pm"] = 61, ["qp"] = true, },
    },
    ["n"] = "Orders from Fildgor",
    ["cn"] = "Plots and Schemes",
  },
```
I just updated that one. I had to be on an alt and doing the quest but now that is properly updated. I won't see that on other characters now and try to find it, looking under a tent on tables or whatever because it's not there unless you are doing the quest.

I found a book that I could update just deconstructing things.

```
  [548] = {
    ["c"] = true,
    ["e"] = {
      [1] = { ["px"] = 0.6405204272, ["py"] = 0.2620299999, ["pm"] = 160, ["zm"] = 61, },
      [1] = { <<-- Going to remove this one
        ["mn"] = 13,
        ["py"] = 0.2620300000,
        ["px"] = 0.6405100000,
        ["pm"] = 61,
      },
    },
    ["n"] = "Note from Thulvald's Logging Camp",
    ["cn"] = "Mysteries and Clues",
  },
``` 
So the second one was the old info. It has the `["mn"]` thing which won't work anymore and the `["pm"]` was for the overland map so I could not see it in Windhelm.

You won't know something like that until you are playing and happen along a situation like that.

**Comment by PorygonA** [comment]
Oh yeah that reminds me, I updated Archdruid Michiel's Orders [7271] on my dev branch. It only shows up on the quest, did I update that right?

**Comment by Sharlikran** [comment]
With Git For Windows you have to copy the changes from the Addons folder to the Git folder. Then commit the changes. Once that is done then I might be able to see it but I'm not sure where to look.

A Fork is `https://github.com/PorygonA/LoreBooks` or a copy of the entire project.

A branch is like a copy of the `.lua` files used for Lorebooks or like how you are in one instance and the group leader is in another instance. You have to travel to the leader to be in the same instance.

The master branch is the main project (group leader) and dev-porygona is like your specific instance.

Until you submit a change to master (which I don't prefer but it's okay) or dev-porygona (which is my preferance) then I can't see the changes.

What did you do after you copied 7271 to your branch, what did you click or do in GitForWindows?

**Comment by PorygonA** [comment]
Yeah I'm on the dev-porygona branch, and did a commit to dev-porygona with summary of "README Typo, Prologue Book"

![image](https://user-images.githubusercontent.com/121068126/209583834-8914630a-32ca-4c4c-b679-6eaad00a5eec.png)

Here

**Comment by Sharlikran** [comment]
![image](https://user-images.githubusercontent.com/1912426/209687916-8a6e1707-fcb8-41ec-93cf-ca390a4944e5.png)

On your fork you have the one branch which is just master.

![image](https://user-images.githubusercontent.com/1912426/209688030-b3e1887c-3152-44a1-aff0-b6b20f26c41f.png)

If I go to my repository I see what I did 5 days ago but not anything you are trying to submit.

So no worries we can get it straightened out. I am happy to help. Keep a backup of your changes somewhere so when I update Lorebooks Minion doesn't overwrite it.

For now copy and paste the change into a comment here while we work on getting everything to work for you.

**Comment by Sharlikran** [comment]
While you are doing that, compare it to my previous mention of the syntax changes.
```
[338] = {
  ["c"] = true,
  ["q"] = 4060,
  ["e"] = {
    [1] = { ["px"] = 0.6301764233, ["py"] = 0.2815471957, ["pm"] = 61, ["qp"] = true, },
  },
  ["n"] = "Orders from Fildgor",
  ["cn"] = "Plots and Schemes",
},
```
Now that is for a regular pin on the overland map with no need for a secondary map ID or secondary x,y coordinates. However, that's the correct way to add a pin that will only show up when the quest is in your quest journal and you have not completed the quest yet.

I have to refine and troubleshoot things today but my intention was to have the book show up if you completed the quest regardless of whether or not the game will display the Lorebook. Because that's what the completed pins are anyway when you toggle them on.

**Comment by PorygonA** [comment]
```
 [7115] = {
    ["c"] = true,
    ["cn"] = "Systres Tomes and Scrolls",
    ["n"] = "Aurelia's Letter",
    ["q"] = 6791,
    ["e"] = {
    },
    ["c"] = false,
      [1] = { ["px"] = 0.0694004026, ["py"] = 0.6027752052, ["pm"] = 2163, ["zm"] = 2114, },
    }
  },


["cn"] = "Systres Tomes and Scrolls",
    ["n"] = "History of House Dufort",
    ["e"] = {
      [1] = { ["px"] = 0.0715032013, ["py"] = 0.5897176011, ["pm"] = 2114, ["r"] = true, },
    },
    ["c"] = false,
    ["c"] = true,
  },
```

**Comment by Sharlikran** [comment]
Close
```
  [7130] = {
    ["c"] = true,
    ["cn"] = "Systres Tomes and Scrolls",
    ["n"] = "History of House Dufort",
    ["e"] = {
      [1] = { ["px"] = 0.0715032013, ["py"] = 0.5897176011, ["pm"] = 2114, },
    },
  },
```
For 7130, use what I have above. You can not have `["c"]` twice because it sets up a routine to tell Lorebooks that there is information to look at under `["e"]`. Also `["r"]` does mean random but it is for Bookshelves because random books appear in them.

`["r"]` has been used incorrectly in the past and I have made changes as I mentioned in the readme.
```
  [653] = {
    ["c"] = true,
    ["cn"] = "Handbills, Posters, and Decrees",
    ["n"] = "Shad Astula Curriculum",
    ["e"] = {
    },
    ["r"] = true,
    ["m"] = {
      [300] = 25,
      [10] = 34,
      [26] = 8,
      [27] = 5,
    },
  },
```
That is the correct use for `["r"]` because it is accompanied by `["m"]` with map IDs and a count. The count is irrelevant. It could as well be just `[300] = true,`

In the above example `["e"]` is empty because that book appears in bookshelves in one of the 4 zones indicated under `["m"]`.

You can have both `["e"]` and `["m"]` now because of recent changes.

```
[7115] = {
    ["c"] = true,
    ["cn"] = "Systres Tomes and Scrolls",
    ["n"] = "Aurelia's Letter",
    ["q"] = 6791,
    ["e"] = {
      [1] = { ["px"] = 0.0694004026, ["py"] = 0.6027752052, ["pm"] = 2163, ["zm"] = 2114, },
    },
  },
```
The line with px, py, and pm etc goes under `["e"]` for the Eidetic Books rather then under `["c"]`.

Use what I have above instead for what you are working on.

**Comment by PorygonA** [comment]
Yeah, History of House Dufort shows up in a bookshelf.

**Comment by Sharlikran** [comment]
I rewrote my comment a few times, so I'd refresh the page.

**Comment by Sharlikran** [comment]
Also do this to resolve your issue with GitForWindows.

Make sure you have a backup of your work outside of the Repository folder.

![image](https://user-images.githubusercontent.com/1912426/209894228-89b449fc-be2a-4513-a197-c3dd12ca7312.png)

Open the Repository in the windows explorer and then

![image](https://user-images.githubusercontent.com/1912426/209894281-330f3e48-f147-4fb6-b19f-f5d15facc290.png)

Go to the parent folder indicated by the red box. The folder before Lorebooks.

![image](https://user-images.githubusercontent.com/1912426/209894307-483c27ba-0054-431b-a5b0-8afe679043c5.png)

Delete your clone or delete the folder in the purple box.

Close and repoen GitForWindows and it will tell you it can not find Lorebooks. Then choose Clone Again. After that switch to your branch and see if you can submit a change.

**Comment by PorygonA** [comment]
After I commit, am I meant to push upstrream?

**Comment by Sharlikran** [comment]
I believe it should say that. I have not done it that way. If it denies you the ability to do that let me know and I have a special command I can have you try.

**Comment by Sharlikran** [comment]
or I can add you hang on

**Comment by Sharlikran** [comment]
In your browser see if you can find where to go to accept an invite to this repository. I believe it will highlight something.

Then try the push to upstream thing you saw.

**Comment by PorygonA** [comment]
Alrighty, accepted and I believe I've pushed upstream.

**Comment by Sharlikran** [comment]
Okay let me fetch your changes

**Comment by Sharlikran** [comment]
![image](https://user-images.githubusercontent.com/1912426/209895417-2a7289d5-0721-4d1b-82b7-6955b5cc23d2.png)

Yeah I can see it now

**Comment by Sharlikran** [comment]
Okay so lets try, so we can avoid the learning curve, to submit one change at a time for one book at a time. Then I can use a Git thing called Cherry-Pick and get the books you changed one at a time once I review them.

**Comment by Sharlikran** [comment]
Question though. You have [7115] showing to be related to the quest 6791, does that book appear only while you are on the quest, after you complete it, or does it appear in that location regardless?

Lorebooks has the ability now to get that nit-picky so unfortunately for quest related books they need to be that nit-picky and accurate.

**Comment by Sharlikran** [comment]
With [7130] you have indicated that the book can be found in that location `OR` in a bookshelf. If I go to that location to check myself will find a book on a table, or a Bookshelf? I have a different system for bookshelves.

**Comment by PorygonA** [comment]
[7115], Aurelia's Letter, shows up on as a certain quest stage during "The Princess Detective", quest 6791. So yes, while on it, but it disappears afterwards.

[7130] shows up on a crate at that location, and at random bookshelves in the zone.

![image](https://user-images.githubusercontent.com/121068126/209897163-64f1409f-84e4-48b4-9572-df7527bf004d.png)

**Comment by Sharlikran** [comment]
Then [7115] needs the `["q"]` tag in order to define the quest number, and in the book entry you need `["qp"] = true` because that indicates while the quest is "In Progress".
```
  [7115] = {
    ["c"] = true,
    ["cn"] = "Systres Tomes and Scrolls",
    ["n"] = "Aurelia's Letter",
    ["q"] = 6791,
    ["e"] = {
      [1] = { ["px"] = 0.0694004026, ["py"] = 0.6027752052, ["pm"] = 2163, ["zm"] = 2114, ["qp"] = true, },
    },
  },
```

Then [7130] is fine if it appears on a crate and in bookshelves.

**Comment by Sharlikran** [comment]
![image](https://user-images.githubusercontent.com/1912426/209897833-80154c6a-5550-4726-841d-544d24d55eb0.png)

Everything in the green box goes in the book entry line under `["e"]`

```
  [1] = { ["px"] = 0.0694004026, ["py"] = 0.6027752052, ["pm"] = 2163, ["zm"] = 2114, ["qp"] = true, },
```

**Comment by Sharlikran** [comment]
Well whatever applies. You don't add things that don't need to be there. For example don't put `["qp"] = false,` if it does not relate to a quest that would be redundant.

**Comment by Sharlikran** [comment]
I'm gonna be unavailable except on Discord for a while.

**Comment by Sharlikran** [comment]
Your version of 7130 was close it needs to be slightly different.
```
  [7130] = {
    ["c"] = true,
    ["cn"] = "Systres Tomes and Scrolls",
    ["n"] = "History of House Dufort",
    ["e"] = {
      [1] = { ["px"] = 0.0715032013, ["py"] = 0.5897176011, ["pm"] = 2114, },
    },
    ["r"] = true,
    ["m"] = {
      [2114] = true,
    },
  },
```

So ["e"] and ["m"] are like the chapter in a book and what is under them are the paragraphs. So there has to be a begin and an end.

```
    ["e"] = {
      <<insert info here>>
    },
```

**Comment by PorygonA** [comment]
Alrighty, updated a ton of Galen stuff.

**Comment by PorygonA** [comment]
Been a while, how do I get the new empty data in my branch?

**Comment by Sharlikran** [comment]
Not sure what you mean by new empty data. Plus I can't really add the default book anymore to the database without it messing up the order of things. ESO doesn't save these arrays in order like 1, 2, 3, 4, 5 it's maybe 4968, xxxx, xxxx, xxxx, 5, xxxx, 1 and whatever. It's all mixed up. Just update or add whatever you need to.

I probably won't add the pull request right away but thanks for that, I appreciate it.

I'll help reset your branch after I get everything finished and then you would just reclone the repository.

**Comment by PorygonA** [comment]
So when Firesong dropped, a bunch of new empty entries for Firesong books got added, like [7467], [7605], [7416] (as an example). I'm wondering how to get the empty ones for Scribes of Fate.

**Comment by Sharlikran** [comment]
> I can't add the default book anymore to the database without it messing up the order of things. ESO doesn't save these arrays in order like 1, 2, 3, 4, 5. They are out of order for example 4968, xxxx, xxxx, xxxx, 5, xxxx, 1 and whatever. It's all mixed up. Just update or add whatever you need to.

I had to quote myself. I can't add that. Add the book when you find it.

**Comment by PorygonA** [comment]
Submitted the Scribes of Fate books, could you also check and upload the things I submitted on Feb 9, 15, and Mar 10?

**Comment by Sharlikran** [comment]
@PorygonA If you have any submissions just go ahead and make them. I wouldn't wait on me. I know I mentioned something I observed a while back but I am bound to forget something I did when I was developing the current syntax for the books. It could have been me so just take note of what I observed and then use the current syntax.

Don't wait for me I may not work on or correct something for a while. I am always going back and forth between projects.

**Comment by PorygonA** [comment]
I've been doing submissions to the [current dev-porygona branch](https://github.com/ESOUIMods/LoreBooks/compare/master...dev-porygona). Last commit I did was a Bastion Nymic book, and I believe I should be all up-to-date with the books. Should be good to update the add-on.

**Comment by Sharlikran** [comment]
@PorygonA I don't think I can use the changes, fiddlesticks. I can't tell but I think you changed the order of the books. Which I don't want the new books added in order nor do I want the order changed. It doesn't need to be in order to work properly.

The reason that I don't want the order changed is that it adds all kinds of mess to everything when I'm checking changes.

**Comment by Sharlikran** [comment]
![image](https://github.com/ESOUIMods/LoreBooks/assets/1912426/c80bbbe0-f02e-4a77-a9db-4e6a6a86fca9)

**Comment by Sharlikran** [comment]
![image](https://github.com/ESOUIMods/LoreBooks/assets/1912426/dfbe84ca-b0ed-4a68-b9d2-7e66043383a9)

**Comment by Sharlikran** [comment]
The game when you export all the books to the saved variables doesn't save anything in order. So I can't import changes from you and then output that in order. Then import yachoor's changes and put that in order and then any new books and put that in order. The my changes and put that in order. The game will randomly put everything in a different order then you put it.

**Comment by Sharlikran** [comment]
Another reason it's not good to reorder things is because Lua doesn't know you want one over the other.
```Lua
[3047] = { -- and a bunch of stuff
}
[3047] = { -- and a bunch of stuff
}
```
The last one will be what the game uses and if things are reordered and then duplicated then maybe I put the changes in the first one. So then my changes don't show up because of the second one.

**Comment by Sharlikran** [comment]
I'm going to check that I didn't do that on my end and mess things up. I'm gonna reinstall with Minion.

**Comment by PorygonA** [comment]
Alrighty, the thing I'm going to do is download the most recent version of the add-on, and re-add the changes I made, minus the reordering

**Comment by Sharlikran** [comment]
No don't do it. Because as I am thinking about it you went to a lot of trouble, probably hours if not days. I'm just going to work with it because I do appreciate it and while I don't normally do things like that if it helps then it helps.

**Comment by Sharlikran** [comment]
I'll figure it out.

**Comment by PorygonA** [comment]
Okay, gotcha. Um. Ignore all the commits I made today then. 😛

It actually surprisingly did not take that long! Few hours-ish.

**Comment by PorygonA** [comment]
Hi;

Sorry to bother you: The changes I submitted on September 21, 2023 was not added with the most recent update.

**Comment by Sharlikran** [comment]
I just updated the edidic memory issue. I haven't had time to look at anything. It wasn't an oversight.

**Comment by PorygonA** [comment]
Gotcha, understood! Again, sorry to bother you, just wanted to be sure.

Have a good one :)

**Comment by Sharlikran** [comment]
No bother, just letting you know. I'm not working on much right now.

**Comment by Sharlikran** [comment]
```
  [5897] = {
    ["c"] = false,
    ["cn"] = "Solitude Soliloquies",
    ["n"] = "Bards College Salskap Tonight!",
    ["q"] = 6482,
    ["e"] = {
    },
  },
```
Changed to
```
  [5897] = {
    ["c"] = true,
    ["cn"] = "Solitude Soliloquies",
    ["n"] = "Bards College Salskap Tonight!",
    ["q"] = 6482,
    ["e"] = {
	  [1] = { ["qp"] = true, },
    },
  },
```
I can't accept that because there is no book location. I understand you need to be on the quest but the whole information needs to be present.

**Comment by PorygonA** [comment]
It goes straight into your inventory during the quest, so it doesn't exactly have a location. It's similar to how the following are:

```
  [7292] = {
    ["c"] = true,
    ["cn"] = "Hearts and Flowers",
    ["n"] = "Letter From Ember",
    ["q"] = 6786,
    ["e"] = {
	  [1] = { ["qc"] = true, },
    },
  },
  [7293] = {
    ["c"] = true,
    ["cn"] = "Hearts and Flowers",
    ["n"] = "Letter From Isobel",
    ["q"] = 6790,
    ["e"] = {
	  [1] = { ["qc"] = true, },
    },
  },
```

These only have "qc"=true as the location, because they too go straight to the inventory.

**Comment by Sharlikran** [comment]
@PorygonA Copy any changes somewhere, delete the git folder for Lorebooks and open Git For Windows. It should say it can't find the folder and when it does clone it again. That will make the dev-PorygonA branch current with master so you can add more changes to it again.

**Comment by PorygonA** [comment]
I have finished that successfully! Submitted one small change to [6252] "The Undaunted Need You!", there were two ["c"], and I removed the one that was set to false and kept the one set to true.

Is deleting the Git and cloning again a task I should do on a semi-regular basis?

**Comment by Sharlikran** [comment]
Not really, maybe after each version or when I suggest it.

**Comment by Sharlikran** [comment]
You didn't rebase your branch for the newer versions. Don't worry about it for now, but in the future.

**Comment by PorygonA** [comment]
It should be good. I deleted, recloned, and it fetched upstream again so it should all be good.

**Comment by PorygonA** [comment]
So, I don't know exactly why GitHub is doing this now.

I try to rebase for U44. I delete the file, reclone, and fetch upstream. Everything is the same as it has been.

For some reason, GitHub saying "D:\Users\PorygonA\Documents\GitHub\Lorebooks" doesn't exist. Create it?" It then refuses to create it.

So GitHub is broken for me somehow.

**Comment by Sharlikran** [comment]
I'll see what I can do to fix it up for you later today. Then you can add some of the provided locations. Thanks.

**Comment by PorygonA** [comment]
Gotcha. I'm not sure what exactly is wrong with it.

Sounds good, I've got my notepad document running with the verified for locations from PTS to live.

**Comment by Sharlikran** [comment]
Try that but from time to time you add things that somehow get removed. Mostly when I do not add the changes because I want to spend more time reviewing what you changed. However, the change doesn't stay on your end sometimes. For now though your branch is just forced to what is current in master. I don't know if that will help.

**Comment by Sharlikran** [comment]
In its current state your branch will make it easy to cherry pick your additions, so it's helpful.

**Comment by PorygonA** [comment]
Seems to work. I've submitted a quick fix for the Gleaner's Claim book, where the book ID was messed up. Did that go through?

**Comment by PorygonA** [comment]
Passing along, submitted a big update. Just wanna make sure it went through.

**Comment by Sharlikran** [comment]
<img width="1346" height="1184" alt="Image" src="https://github.com/user-attachments/assets/0aa45f5a-4269-424f-a431-8e29d46bb737" />

[2693] = { ["cn"] = "Secrets de Raidelorn", ["n"] = "Ordres de la régente Elska", ["e"] = { [1] = { ["px"] = 0.3805616131, ["py"] = 0.3248823972, ["pm"] = 1115, ["d"] = true, }, }, }, { ["sm"] = 1115, ["pnx"] = 0.7103614211, ["pny"] = 0.5262650847 }, -- craglorn/serpentsnest_base_0

That is a correction for that book

<img width="797" height="795" alt="Image" src="https://github.com/user-attachments/assets/ee05fc7a-c266-4852-8df2-9ce871226a14" />

[1039] = { ["cn"] = "Notes de recherche", ["n"] = "Le marché de Scinderacine", ["e"] = { [1] = { ["px"] = 0.4498928126, ["py"] = 0.7431668070, ["pm"] = 283, ["d"] = true, }, }, }, { ["sm"] = 283, ["pnx"] = 0.4771356881, ["pny"] = 0.7033291459 }, -- grahtwood/rootsunder_base_0

Another correction

<img width="447" height="612" alt="Image" src="https://github.com/user-attachments/assets/ce9cbbd2-e694-4793-aac8-b2431be6e735" />

[5451] = { ["cn"] = "Archives d'Anequina", ["n"] = "Mojha est une andouille", ["e"] = { [1] = { ["px"] = 0.5066976294, ["py"] = 0.6887576175, ["pm"] = 1649, ["d"] = true, }, }, }, { ["sm"] = 1649, ["pnx"] = 0.4108352065, ["pny"] = 0.5178706050 }, -- elsweyr/sunspireoverworld_base_0

That's a correction or maybe relocation

<img width="406" height="396" alt="Image" src="https://github.com/user-attachments/assets/51b6c46d-1de9-410b-88f8-0112ca208f39" />

[8412] = { ["cn"] = "Manuscrits introuvables", ["n"] = "Tome temporel de Gastecairn", ["e"] = { [1] = { ["px"] = 0.0840212003, ["py"] = 0.6517184142, ["pm"] = 1397, ["d"] = true, }, }, }, { ["sm"] = 1397, ["pnx"] = 0.2003379911, ["pny"] = 0.4404191971 }, -- summerset/sum_karnwasten_base_0

## Maybe Books or Bookshelves

[7212] = { ["e"] = { [1] = { ["px"] = 0.0599280016, ["py"] = 0.5862488019, ["pm"] = 2114, }, }, }, { ["pnx"] = 0.3332854509, ["pny"] = 0.4797705412 }, -- Le Tournoi du Saphir étincelant, systres/u34_systreszone_base_0

[7206] = { ["e"] = { [1] = { ["px"] = 0.0539888019, ["py"] = 0.5872348032, ["pm"] = 2114, }, }, }, { ["pnx"] = 0.2078450918, ["pny"] = 0.5005956292 }, -- L'écologie des ornaugs, systres/u34_systreszone_base_0

[7213] = { ["e"] = { [1] = { ["px"] = 0.0587376020, ["py"] = 0.5983596031, ["pm"] = 2114, }, }, }, { ["pnx"] = 0.3081433177, ["pny"] = 0.7355597615 }, -- Page du journal de la guérisseuse, systres/u34_systreszone_base_0

[7216] = { ["e"] = { [1] = { ["px"] = 0.0740624016, ["py"] = 0.5812220021, ["pm"] = 2114, }, }, }, { ["pnx"] = 0.6318145990, ["pny"] = 0.3736007512 }, -- L'histoire des Systres : volume 7, systres/u34_systreszone_base_0

verify those 4 please and make sure they are not bookshelves.

**Comment by Sharlikran** [comment]
@PorygonA can you look at those please before I update Lorebooks?

**Comment by PorygonA** [comment]
[2693] and [1039] are good and fixed.

I can't do [5451], I do not have the time to run Sunspire.

[8412] is basically already where the book is. I'll change it to move it slightly closer.

[7212], [7206], [7213], and [7216] are all legitimate books. I'll enter in these new coordinates, and submit.

**Comment by Sharlikran** [comment]
[7212], [7206], [7213], and [7216]

When I asked are they books but I mean are the coordinates a bookshelf this person didn't know what I meant originally when I added the bookshelves and started mass giving me bookshelf data which I wasn't going to add and I had to remake the data gather slash commands.

So if you checked already that that's all I wondered. Thanks.

**Comment by PorygonA** [comment]
Ah, got it. Yeah those are all individual books and not bookshelves.

All is good to update :)
